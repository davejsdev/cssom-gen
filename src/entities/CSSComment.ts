export type CSSCommentInlineCSSText = `/* ${string} */`;
export type CSSCommentBlockCSSText = `/**\n${string}\n */`;
export type CSSCommentCSSText = CSSCommentInlineCSSText | CSSCommentBlockCSSText;

export type CSSCommentType = 'inline' | 'block';
export interface CSSCommentOptions {
  type: CSSCommentType;
}

/**
 * A comment in a CSS stylesheet.
 */
export class CSSComment {
  static defaultOptions: CSSCommentOptions = {
    type: 'inline',
  };

  value: string;
  type: CSSCommentType;

  constructor(value: string, options: CSSCommentOptions = CSSComment.defaultOptions) {
    this.value = value;
    this.type = options.type;
  }

  get cssText(): CSSCommentCSSText {
    if (this.type === 'inline') {
      return `/* ${this.value} */`;
    }

    if (this.type === 'block') {
      // Format the value as a block comment
      const lines = this.value.split('\n');
      const indentedLines = lines.map((line) => ` * ${line}`);
      const formattedLines = indentedLines.join('\n');

      return `/**\n${formattedLines}\n */`;
    }

    throw new Error('Invalid comment type');
  }
}
