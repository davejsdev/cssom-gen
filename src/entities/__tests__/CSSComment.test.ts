import { describe, expect, it } from 'vitest';
import { CSSComment } from '../CSSComment.js';

describe('CSSComment', () => {
  it('should create a comment', () => {
    const comment = new CSSComment('This is a comment');
    expect(comment.cssText).toMatchSnapshot();
  });
  it('should create a block comment', () => {
    const comment = new CSSComment('This is a comment', { type: 'block' });
    expect(comment.cssText).toMatchSnapshot();
  });
});
