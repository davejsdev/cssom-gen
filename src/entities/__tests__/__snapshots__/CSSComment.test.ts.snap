// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CSSComment > should create a block comment 1`] = `
"/**
 * This is a comment
 */"
`;

exports[`CSSComment > should create a comment 1`] = `"/* This is a comment */"`;
